### PR TITLE
Adding the GSG adoc version and changes to the workflow.

### PR DESCRIPTION
**Pull request by Preeticp**
## What is the purpose of this change? What does it change?

- This PR converts the GSG to adoc format.
- It also makes changes to the workflow as pointed out by @surajnarwade and @anmolbabu

### CODE REVIEW

**Comment by Preeticp** [comment]
@surajnarwade @cdrage please review the change in the workflow. The adoc related changes were approved by @cdrage and @kadel in #1553, however I created this fresh one and will be closing the older one.
@cdrage, please note as per my discussions with Robert (@rkratky) he will take care of making script changes to the adocs so that they render appropriately for the website.

File: docs/getting-started.adoc [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+----
+$ odo login -u developer -p developer
+----
++
+[NOTE]
+====
+In order to make full use of `odo` functionality, it is recommended to enable the OpenShift link:https://docs.openshift.com/container-platform/3.11/architecture/service_catalog/index.html[service catalog]. Use `minishift` 1.30 or latest to enable this:
+
+----
+$ MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags "--enable=*,service-catalog,automation-service-broker,template-service-broker"
```
Comment (surajnarwade) [inline docs/getting-started.adoc]: instead of rewriting, just point it to `service catalog` section later in this doc

Comment (Preeticp) [reply docs/getting-started.adoc]: @surajnarwade here we say 'Use `minishift` 1.30 or latest to enable this:'
In the main section on line 135, it says 1.22 or above, can you confirm what version we want to ask the user to use?

Comment (surajnarwade) [reply docs/getting-started.adoc]: `1.22 and above` is fine

Comment (Preeticp) [reply docs/getting-started.adoc]: Fixed

**Comment by kadel** [comment]
I think that this is ok for now. We will have to update it once we finish with all "new workflow" changes

/approve

**Comment by Preeticp** [comment]
> I think that this is ok for now. We will have to update it once we finish with all "new workflow" changes
@kadel agree, as per my discussions with @surajnarwade once the changes are in just add the `Needs docs` label to the issue/PR and ping us, we will work on it.

**Pull request merged**